src/components/dialogs/two-windings-transformer/two-windings-transformer-creation-dialog.js

```diff
@@ -0,0 +1,1100 @@
+import { useSnackbar } from 'notistack';
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect, useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { useParams } from 'react-router-dom';
+import {
+    displayErrorMessageWithSnackbar,
+    useIntlRef,
+} from '../../../utils/messages';
+import { createTwoWindingsTransformer } from '../../../utils/rest-api';
+import {
+    useButtonWithTooltip,
+    useDoubleValue,
+    useEnumValue,
+    useIntegerValue,
+    useRegulatingTerminalValue,
+    useTextValue,
+} from '../inputs/input-hooks';
+import EquipmentSearchDialog from '../equipment-search-dialog';
+import { useFormSearchCopy } from '../form-search-copy-hook';
+import TwoWindingsTransformerPane from './two-windings-transformer-pane';
+import { useInputForm } from '../inputs/input-hooks';
+import RatioTapChangerPane from './ratio-tap-changer-pane';
+import PhaseTapChangerPane from './phase-tap-changer-pane';
+import { useConnectivityValue } from '../connectivity-edition';
+import {
+    ActivePowerAdornment,
+    AmpereAdornment,
+    filledTextField,
+    MVAPowerAdornment,
+    OhmAdornment,
+    MicroSusceptanceAdornment,
+    VoltageAdornment,
+    sanitizeString,
+} from '../dialogUtils';
+import { REGULATION_MODES } from '../../network/constants';
+import { useBooleanValue } from '../inputs/boolean';
+import { EQUIPMENT_TYPE } from '@gridsuite/commons-ui';
+
+export const PHASE_TAP = 'dephasing';
+export const RATIO_TAP = 'ratio';
+
+/**
+ * Dialog to create a two windings transformer in the network
+ * @param {Boolean} open Is the dialog open ?
+ * @param {EventListener} onClose Event to close the dialog
+ * @param voltageLevelOptionsPromise Promise handling list of voltage level options
+ * @param currentNodeUuid : the node we are currently working on
+ * @param editData the data to edit
+ */
+const TwoWindingsTransformerCreationDialog = ({
+    editData,
+    open,
+    onClose,
+    voltageLevelOptionsPromise,
+    voltageLevelsEquipmentsOptionsPromise,
+    currentNodeUuid,
+}) => {
+    const studyUuid = decodeURIComponent(useParams().studyUuid);
+
+    const intlRef = useIntlRef();
+    const intl = useIntl();
+
+    const { enqueueSnackbar } = useSnackbar();
+
+    const characteristicsInputForm = useInputForm();
+    const ratioTapInputForm = useInputForm();
+    const phaseTapInputForm = useInputForm();
+
+    const [formValues, setFormValues] = useState({});
+
+    const [ratioTapRows, setRatioTapRows] = useState([]);
+
+    const [phaseTapRows, setPhaseTapRows] = useState([]);
+
+    const [dialogWidth, setDialogWidth] = useState('sm');
+
+    const equipmentPath = '2-windings-transformers';
+
+    const [tabIndex, setTabIndex] = useState(0);
+
+    const clearValues = () => {
+        setFormValues(null);
+    };
+
+    const [isCopy, setIsCopy] = useState(false);
+
+    const [creationError, setCreationError] = useState(undefined);
+
+    /**
```

Comment:
VSCode minimpa effect ? ;-)

Reply:
I'm not sure I get it, you suggest to remove the comment or replace with some sort of visual anchor in VScode ?

Reply:
No, just wondering why such a big block of non-code. And my privileged hypothesis about it was to have it outstanding in the minimap.
But, if you don't mind, remove the lines with just a star on it ;-)

Reply:
✓